packages/web/src/common/store/cache/tracks/sagas.ts

```diff
@@ -90,6 +90,34 @@ function* watchAdd() {
   )
 }
 
+function* watchUpdate() {
+  yield* takeEvery(
+    cacheActions.UPDATE,
+    function* (action: ReturnType<typeof cacheActions.update>) {
+      const getFeatureEnabled = yield* getContext('getFeatureEnabled')
+      const isStreamPrefetchEnabled = yield* call(
+        getFeatureEnabled,
+        FeatureFlags.PREFETCH_STREAM_URLS
+      )
+      if (!isStreamPrefetchEnabled) return
+
+      if (action.kind === Kind.TRACKS) {
+        // Check for tracks with any changed access. If so we need to update the prefetched stream url
+        const tracksWithChangedAccess = action.entries
+          .filter((track) => track.metadata.access.stream !== undefined)
```

Comment:
Stream access changes seems to me the only reason we would want to refetch the stream url? I can't think of anything else